picard/util/checkupdate.py

```diff
@@ -0,0 +1,177 @@
+
+        if self._available_versions['stable']['tag']:
+            # Release information already acquired from specified website api.
+            self._display_results()
+        else:
+            # Gets list of releases from specified website api.
+            self._query_available_updates(callback=callback)
+
+    @property
+    def available_versions(self):
+        '''Provide a list of the latest version tuples for each update type.'''
+        return self._available_versions
+
+    def _query_available_updates(self, callback=None):
+        '''Gets list of releases from specified website api.'''
+        log.debug("Getting release information from {host_url}".format(host_url=PLUGINS_API['host'],))
+        self.tagger.webservice.get(
+            PLUGINS_API['host'],
+            PLUGINS_API['port'],
+            PLUGINS_API['endpoint']['releases'],
+            partial(self._releases_json_loaded, callback=callback),
+            parse_response_type=None,
+            priority=True,
+            important=True
+        )
+
+    def _releases_json_loaded(self, response, reply, error, callback=None):
+        '''Processes response from specified website api query.'''
+        if error:
+            log.error(_("Error loading releases list: {error_message}").format(error_message=reply.errorString(),))
+            if self._show_always:
+                QMessageBox.information(
+                    self._parent,
+                    _("Picard Update"),
+                    _("Unable to retrieve the latest version information from the website.\n(https://{url}{endpoint})").format(
+                        url=PLUGINS_API['host'],
+                        endpoint=PLUGINS_API['endpoint']['releases'],
+                    ),
+                    QMessageBox.Ok, QMessageBox.Ok)
+        else:
+            self._available_versions = load_json(response)['versions']
+            for key in self._available_versions:
+                log.debug("Version key '{version_key}' --> {version_information}".format(
+                    version_key=key, version_information=self._available_versions[key],))
+            self._display_results()
+        if callback:
+            callback(not error)
+
+    def _display_results(self):
+        # Display results to user.
+        key = ''
+        high_version = PICARD_VERSION
+        for test_key in PROGRAM_UPDATE_LEVELS:
+            test_version = self._available_versions[PROGRAM_UPDATE_LEVELS[test_key]['name']]['version']
+            if self._update_level >= test_key and  compare_version_tuples(high_version, test_version) > 0:
+                key = PROGRAM_UPDATE_LEVELS[test_key]['name']
+                high_version = test_version
+        if key:
+            if QMessageBox.information(
+                self._parent,
+                _("Picard Update"),
+                _("A new version of Picard is available.\n\n"
+                  "Old version: {picard_old_version}\n"
```

Comment:
s/Old version/Current version/?

Reply:
How about 'Your version' or 'This version' instead?  When I hear the term 'Current version' I automatically think of the latest version.